packages/medusa-plugin-algolia/src/services/algolia.ts

```diff
@@ -105,8 +125,14 @@ class AlgoliaService extends SearchService {
    * - additionalOptions contain any provider specific options
    * @return {*} - returns response from search engine provider
    */
-  async search(indexName, query, options) {
+  async search(
+    indexName: string,
+    query: string,
+    options: SearchOptions & Record<string, unknown>
+  ) {
     const { paginationOptions, filter, additionalOptions } = options
+
+    // fit our pagination options to what Algolia expects
     if ("limit" in paginationOptions) {
```

Comment:
suggestion: Here we are just checking that the limit prop is in the object. should we check that it != null since the prop can be provided but it can be null or undefined if used in js?